nibabel/freesurfer/io.py

```diff
@@ -372,11 +394,101 @@ def read_annot(filepath, orig_ids=False):
     return labels, ctab, names
 
 
-def write_annot(filepath, labels, ctab, names):
-    """Write out a Freesurfer annotation file.
+def _read_annot_ctab_old_format(fobj, n_entries):
+    """Read in an old-style Freesurfer colour table from `fobj`.
+
+    This function is used by :func:`read_annot`.
+
+    Parameters
+    ----------
+
+    fobj : file-like
```

Comment:
File-like usually means it can be a filename.  I think that a filename will in fact work here, but will give the wrong result, because the position will always be 0 when doing the read.  Maybe worth a check and comment?

Reply:
You're right - a file name would work. I've changed the docs to specify `file`, rather than `file-like`. 

But I'm not quite sure how to check this, or whether it is necessary (given that these are private functions). I could check the seek position, but that would need both private functions to be passed the number of vertices that were read in beforehand, which I feel is getting a bit over the top. 

What do you think?

Reply:
Does it work though?   It looks to me as if you would pass the filename to each call, and the reading would start at position 0 each time?

It's fine to just say file-like object but not filename in the docstring, and test for the `read` attribute with an assert at the top - if that's the only interface you're using - I think it is?

Reply:
Well, when I said "a file name would work", I mean it would be accepted and probably wouldn't crash. But the data read in would be wrong because, as you say, the seek location will be 0. 

I'm not not directly using any methods on the file object - all of the reading is performed by `numpy.fromfile`. But it can't hurt to check that the file-like has a `read` method - will add it in.